interface/controller/pokemon.go

```diff
@@ -59,3 +61,65 @@ func (ic PokemonController) ActionPostByID(c Context) error {
 		http.StatusCreated,
 	)
 }
+
+func adaptFilters(c Context) (map[string]interface{}, error) {
+	filters := make(map[string]interface{})
+
+	qp := c.QueryParams()
+
+	if qp["type"][0] == "odd" {
+		filters["id"] = func(id int) bool {
+			return id%2 == 1
+		}
+	} else if qp["type"][0] == "even" {
+		filters["id"] = func(id int) bool {
+			return id%2 == 0
+		}
+	} else {
+		filters["id"] = func(id int) bool {
+			return true
+		}
+	}
+
+	if qp["items"] != nil {
+		limit, err := strconv.Atoi(qp["items"][0])
+
+		if err != nil {
+			return nil, customErrors.ErrInvalidRequest{Message: "Invalid 'items' queryParam", Err: err}
+		}
+
+		filters["limit"] = limit
+	} else {
+		filters["limit"] = 10
+	}
+
+	if qp["items_per_workers"] != nil {
+		wj, err := strconv.Atoi(qp["items_per_workers"][0])
+
+		if err != nil {
+			return nil, customErrors.ErrInvalidRequest{Message: "Invalid 'items_per_workers' queryParam", Err: err}
+		}
+		filters["workerJobs"] = wj
+	} else {
+		filters["workerJobs"] = 4
+	}
+
+	return filters, nil
+}
+
+// ActionGetByFilters calls GetPokemonsByFilters.
+func (ic PokemonController) ActionGetByFilters(c Context) error {
+	filters, err := adaptFilters(c)
+
+	if err != nil {
+		return responseErrorJSON(c, err)
+	}
+
+	result, err := ic.Usecase.GetPokemonsByFilters(filters)
+
+	if err != nil {
+		return responseErrorJSON(c, err)
+	}
```

Comment:
Error check is performed in the same code block, immediately after the err assignation.
```suggestion
	filters, err := adaptFilters(c)
	if err != nil {
		return responseErrorJSON(c, err)
	}

	result, err := ic.Usecase.GetPokemonsByFilters(filters)
	if err != nil {
		return responseErrorJSON(c, err)
	}
```